apis/v1alpha1/backendpolicy_types.go

```diff
@@ -31,16 +31,25 @@ import (
 // implementation specific may be represented with similar implementation
 // specific custom resources.
 type BackendPolicy struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   BackendPolicySpec   `json:"spec,omitempty"`
+	// Spec defines the desired state of BackendPolicy.
+	//
+	// +optional
```

Comment:
What's the utility of a resource with an empty spec?

Reply:
Same as https://github.com/kubernetes-sigs/service-apis/pull/538/files#r571116077.

A potential use case is to use api defaulting for spec. A resource with all defaults can be created as such:
```
kind: Gateway
apiVersion: networking.x-k8s.io/v1alpha1
metadata:
  name: example
```

and will result in:
```
kind: Gateway
apiVersion: networking.x-k8s.io/v1alpha1
metadata:
  name: example
spec:
  gatewayClassName: default
  listeners:
  - protocol: HTTP
    port: 80
    routes:
      kind: HTTPRoute
      selector:
        matchLabels:
          app-type: http-exposed
      namespaces:
        from: "Same"
```

This could simplify the user experience for the basic http use case.

__Note:__ All these defaults above do not exist today.

Reply:
I wonder whether this UX would be better in a kubectl plugin? I'm a bit skeptical that defaulting at this layer is the right approach.